### ISSUE TITLE
Write a Rollout Sampling Utility for PPO Agents and add features affect generated distribution.

### ISSUE THREAD

**Issue by jbloomAus** [jbloomAus/DecisionTransformerInterpretability#46]
#43 was insufficient to create an optimal behaviour agent with the current rollout data. 

To falsify the hypothesis that our failure to generate good trajectories (meet the criteria of both solving the task and being reasonable RTG modulated, at all, calibration unimportant for the time being) is a function of the data generating distribution with the current rollout methodology and PPO agent, I would like upgrade our ability to sample from PPO agents. 

This will include:
1.  Adding an option to, rather than randomly sample from the posterior distribution indicated by the logits, sample deterministically from the best action.
2. Adding an option to sample with temperature. This will enable us to calibrate the right amount of randomness. 
3. Adding an option to sample with from anything except the maximal logit (unlike temperature, this will reliably produce anti-good trajectories rather than consistently good trajectories). 

Furthermore, the utility for this should also provide some feedback on the sampled trajectories (for example, average RTG achieved, RTG distribution, trajectory length distribution etc). 

Tasks:

- [x] Write the end-end utility (ie, wrap what we have currently in a runner)
- [x] Add each of the options above with tests
    - [x] Deterministic Sampling
    - [x] Temperature Sampling
    - [x] Bottom-k sampling
- [x] Make it possible to select more than of these and specify the number of rollouts to generate with each.
- [ ] Make it possible to view some example rollouts when doing so/also have some performance metrics.

**Comment by jbloomAus** [comment]
Spinning of last one into a card for upgrading the collect demonstrations workflow.
